bundles/CoreBundle/Command/MaintenanceCommand.php

```diff
@@ -80,8 +86,33 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $validJobs = $this->getArrayOptionValue($input, 'job');
         $excludedJobs = $this->getArrayOptionValue($input, 'excludedJobs');
-
-        $this->maintenanceExecutor->executeMaintenance($validJobs, $excludedJobs, (bool) $input->getOption('force'));
+        $async = (bool)$input->getOption('async');
+
+        $this->maintenanceExecutor->executeMaintenance(
+            $validJobs,
+            $excludedJobs,
+            (bool)$input->getOption('force'),
+            $async
+        );
+
+        if (!$async) {
+            trigger_deprecation(
+                'pimcore/pimcore',
+                '10.2',
+                'Running Maintenance Command without --async and not having the messenger consume message yourself is deprecated and will be removed from Pimcore in 11.',
+                __CLASS__
+            );
+
+            $command = $this->getApplication()->find('messenger:consume');
+
+            $arguments = [
+                'receivers' => ['pimcore_core'],
```

Comment:
```suggestion
                'receivers' => ['pimcore_core', 'pimcore_maintenance'],
```
We should also run the `pimcore_maintenance` immediately.

Reply:
this runs immediately. checkout the task executor

Reply:
nevermind, just read the other comment ;)

Reply:
I thought we could even let `messenger:consume` process the maintenance tasks if running in legacy mode, since we're calling `messenger:consume` anyway. But yes, the current implementation works as well.